torch/csrc/stable/accelerator.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <torch/csrc/inductor/aoti_runtime/utils.h>
+
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+
+namespace torch::stable::accelerator {
+
+namespace {
+inline void delete_device_guard(void* ptr) {
+  AOTI_TORCH_ERROR_CODE_CHECK(
+      aoti_torch_delete_device_guard(reinterpret_cast<DeviceGuardHandle>(ptr)));
+}
+
+} // namespace
+
+using DeviceIndex = int8_t;
+using StreamId = int64_t;
+class DeviceGuard {
```

Comment:
something that I'm not sure about -- do the copy / move semantics need to match the real device guard since this just a wrapper that is holding a `std::unique_ptr` to `DeviceGuardOpaque`?

Reply:
It doesn't need to copy the existing DeviceGuard. For example, I agree with deleting the default constructor for this API. We can disallow copy and move if it's easiest to maintain anyway.